mkosi/installer/dnf.py

```diff
@@ -34,8 +34,13 @@ def find_rpm_gpgkey(state: MkosiState, key: str, url: str) -> str:
 
 
 def dnf_executable(state: MkosiState) -> str:
+    # Allow the user to override autodetection with an environment variable
+    dnf = os.getenv("MKOSI_DNF")
```

Comment:
Let's fetch this from `state.config.environment` instead, and propagate `MKOSI_DNF` into `state.config.environment` in `load_environment()`. That allows `MKOSI_DNF` to also be configured using `Environment=`.

Reply:
Updated like this.